src/main/java/com/googlecode/lanterna/gui2/TextBox.java

```diff
@@ -314,31 +368,32 @@ public TerminalPosition getCaretPosition() {
      * @return Itself
      */
     public synchronized TextBox setCaretPosition(int column) {
-        return setCaretPosition(getCaretPosition().getRow(), column);
+        return setCaretPosition(caretPosition.withColumn(column));
     }
 
     /**
      * Moves the text caret position to a new position in the {@link TextBox}. For single-line {@link TextBox}:es, the
      * line component is not used. If one of the positions are out of bounds, it is automatically set back into range.
      * @param line Which line inside the {@link TextBox} to move the caret to (0 being the first line), ignored if the
      *             {@link TextBox} is single-line
-     * @param column  What column on the specified line to move the text caret to (0 being the first column)
+     * @param position What row & column to move the text caret to (0 being the first column)
      * @return Itself
      */
-    public synchronized TextBox setCaretPosition(int line, int column) {
```

Comment:
This is a breaking change, please keep the original signature (you can forward the call into your new signature)

Reply:
Will clean this up once revisiting this PR.